### PR TITLE
Handle fareproduct use IDs correctly

### PR DESCRIPTION
**Pull request by daniel-heppner-ibigroup**
This fixes fares being counted multiple times when a fare module doesn't provide per-leg fares.

### CODE REVIEW

**Comment by leonardehrenfried** [comment]
I support this change as it looks correct to me.

File: packages/core-utils/src/itinerary.ts [hunk 1]
```diff
@@ -566,7 +566,7 @@ export function getLegCost(
   leg: Leg,
   mediumId: string | null,
   riderCategoryId: string | null
-): { price?: Money; transferAmount?: Money | undefined } {
+): { price?: Money; transferAmount?: Money | undefined; useId?: string } {
```
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/itinerary.ts:569]: Maybe rename `useId` to `productId`?

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM, I would rename `useId`  as indicated but the other reviewer can weigh in.

File: packages/core-utils/src/itinerary.ts (hunk 1 above)
Comment (leonardehrenfried) [reply packages/core-utils/src/itinerary.ts:569]: The thing is, it's _not_ the product id. It is an id that refers to the "instance" of the product, so for example a day pass that you use several times, hence "useId".

The product id globally identifies the product that you're buying and probably has an id like `atl:daypass` but the "useId" is an identifier local to the itinerary allows you to deduplicate the day pass so you know you don't have to buy three passes but it's applicable to all of the legs.

Comment (binh-dam-ibigroup) [reply packages/core-utils/src/itinerary.ts:569]: Okay. How about `ticketId`?

Comment (daniel-heppner-ibigroup) [reply packages/core-utils/src/itinerary.ts:569]: How about `productUseId` to more closely match what it's called in the backend?

Comment (binh-dam-ibigroup) [reply packages/core-utils/src/itinerary.ts:569]: Sure

**Review by miles-grant-ibigroup — APPROVED** [review]
Well done figuring out the fix! No notes from me, I think all variable names are fine.

**Pull request merged**